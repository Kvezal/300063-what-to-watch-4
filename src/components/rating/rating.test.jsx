import React from "react";
import render from "react-test-renderer";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Rating from "./rating";


Enzyme.configure({
  adapter: new Adapter(),
});

describe(`RatingComponent`, () => {
  test(`should render component`, () => {
    const tree = render.create(
        <Rating
          value="2"
          onChange={() => {}}
        />
    )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test(`should create component`, () => {
    const ratingComponent = shallow(
        <Rating
          value="2"
          onChange={() => {}}
        />
    );
    const rating = ratingComponent.find(`.rating`);
    expect(rating).toHaveLength(1);
  });

  test(`should set star count`, () => {
    const starCount = 10;
    const ratingComponent = shallow(
        <Rating
          starCount={starCount}
          value="2"
          onChange={() => {}}
        />
    );
    const starts = ratingComponent.find(`.rating__label`);
    expect(starts).toHaveLength(starCount);
  });

  test(`click on star should call onChange`, () => {
    const onChange = jest.fn();
    const ratingComponent = shallow(
        <Rating
          value="2"
          onChange={onChange}
        />
    );
    const starts = ratingComponent.find(`.rating__input`);
    starts.at(3).simulate(`change`);
    expect(onChange).toBeCalledTimes(1);
  });
});
