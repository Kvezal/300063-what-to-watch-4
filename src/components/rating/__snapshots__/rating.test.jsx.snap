// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingComponent should render component 1`] = `
<div
  className="rating"
>
  <div
    className="rating__stars"
  >
    <input
      checked={false}
      className="rating__input"
      id="star-1"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="1"
    />
    <label
      className="rating__label"
      htmlFor="star-1"
    />
    <input
      checked={true}
      className="rating__input"
      id="star-2"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="2"
    />
    <label
      className="rating__label"
      htmlFor="star-2"
    />
    <input
      checked={false}
      className="rating__input"
      id="star-3"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="3"
    />
    <label
      className="rating__label"
      htmlFor="star-3"
    />
    <input
      checked={false}
      className="rating__input"
      id="star-4"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="4"
    />
    <label
      className="rating__label"
      htmlFor="star-4"
    />
    <input
      checked={false}
      className="rating__input"
      id="star-5"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="5"
    />
    <label
      className="rating__label"
      htmlFor="star-5"
    />
  </div>
</div>
`;
